src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.Comments.cs

```diff
@@ -39,18 +39,27 @@ public partial class zip_CreateTests : ZipFileTestBase
             // Emoji not supported by latin1
             Create_Comment_EntryName_Encoding_Internal(Utf8AndLatin1FileName, originalComment, expectedComment, Encoding.Latin1);
 
```

Comment:
We're going to need at least one more test that confirms that if we open an existing archive where the general purpose bit flag is set for one entry, but we open the archive specifying an encoding, we will now need to ignore the archive encoding and prioritize the general purpose bit flag value. Can you please add it? The logic would be very similar to the one in the `Create_Comment_EntryName_Encoding_Internal` method: you create the archive yourself. Just make sure that when you open the archive again, you specify an `entryNameEncoding`, because we need to confirm that when reading the entry, we will prioritize the general purpose bit flag value over the archive's specified encoding.

Reply:
I think this is covered by `Create_Comment_Utf8EntryName_Utf8Encoding_Prioritised` now. I've also added three more, which should cover various scenarios when the encoding is different and the UTF8 bit is unset, when it defaults to UTF8 if unspecified upon read and the UTF8 bit is unset, and when varying encodings between creation and verification can result in an invalid string.